### ISSUE TITLE
Adjust Schedule for late check in 

### ISSUE THREAD

**Issue by burnclouds** [jeffpiazza/derbynet#282]
I know this is a newer feature and it's great when it works. However, it seems like there are times when adjusting. The schedule takes a while and there's no feedback. The button looks great compared to the unschedule buttons so with no feedback it's hard to tell if it's actually working and on occasion I have gotten an Ajax 500 error while waiting on it to process, especially if I get impatient and try and do something else while it's rescheduling.

In short, I don't know if there's something wrong with my setup. If I'm doing something wrong or simply feedback is needed.

**Comment by burnclouds** [comment]
I'm on Debian and can get you any logs you like. If you know where they typically are located that would be helpful. I can clear them, replicate the issue, then send you the logs. I think it's really just a UI issue, I don't know that I need to wait, so I hit the button again, causing the issue.

**Comment by jeffpiazza** [comment]
/var/log/nginx/error.log and /var/log/nginx/access.log.  If there is one, /var/log/php.log or the like.

**Comment by burnclouds** [comment]
[access.log](https://github.com/jeffpiazza/derbynet/files/13942345/access.log)
[error.log](https://github.com/jeffpiazza/derbynet/files/13942348/error.log)
[20240115-1350.tar.gz](https://github.com/jeffpiazza/derbynet/files/13942364/20240115-1350.tar.gz)

This zip file includes the derbynet "about" page with nginx and PHP version information. It also includes a HAR/XML file from the Network DevTool.

**Comment by burnclouds** [comment]
I'm not sure why php8.2-fpm.log wasn't generated before. but while playing with this schedule adjust feature when getting the ajax 500 error I noticed i started getting this entry in the php-fpm log. There is nothing else relevant or errors reported in the php-fpm log.

[15-Jan-2024 14:21:51] WARNING: [pool www] server reached pm.max_children setting (5), consider raising it

**Comment by jeffpiazza** [comment]
From the error log:

PHP Fatal error:  Maximum execution time of 30 seconds exceeded in /var/www/html/derbynet/ajax/action.schedule.reschedule.inc on line 254

To adjust the timeout, edit /etc/php/8.2/fpm/php.ini, uncomment the max_execution_time variable and set it to something like 600.  Also update the fastcgi_read_timeout value /etc/nginx/http.d/default.conf.  Finally, reload the config file for nginx (nginx -s reload).

Here's a script to do that, if you prefer:
```

TIMEOUT=600

for INI in `find /etc/php* -name php.ini` ; do
    sed -i \
        -e "s/^ *max_execution_time *= *[0-9][0-9]*/max_execution_time = $TIMEOUT/" \
        $INI
done

sed -i \
    -e "/fastcgi_read_timeout/ s/.*/fastcgi_read_timeout $TIMEOUT;/" \
    /etc/nginx/http.d/default.conf

nginx -s reload
```

**Comment by burnclouds** [comment]
I'll try that now. Also i just noticed that when adjusting the schedule it's adding ALL racers that are not checked-in in addition to the late checkins. 
![image](https://github.com/jeffpiazza/derbynet/assets/9850267/a4860234-5260-43bb-821f-2722248aa84f)

**Comment by jeffpiazza** [comment]
Probably fastcgi.conf is where you'd find the timeout value, but I can't be sure.  You might grep for fastcgi_read_timeout among the /etc/nginx files.

Including not-checked-in racers is definitely a bug, with a fix on the way.  That may well also be why the reschedule operation is taking so long for you.

**Comment by icyspace** [comment]
I am able to recreate the same issue on docker.  

- When I test 10 Racers the update works, when I get above 30 it times out. 
- I make the script changes recommend above and that allows it to finish.  I am assuming that configuration change would need to be applied during the docker build process?  I was going to look at that in a bit.  
- I am also seeing the bug you mention and see every one in that schedule that is not checked in is checked in. 
- I agree with @burnclouds that there should be some indication that the button was clicked.  One thing you can do to see that it's working is have the checkin page open and you will see each request processing in green message on the bottom of that page.   

Happy to help test the fix when it's ready.
